src/mlpack/methods/dbscan/dbscan.hpp

```diff
@@ -51,6 +51,10 @@ template<typename RangeSearchType = RangeSearch<>,
 class DBSCAN
 {
  public:
+   //! Easy access to the MatType.
+  typedef typename RangeSearchType::Mat MatType;
+  //! Easy access to Element Type of the matrix.
+  typedef typename RangeSearchType::Mat::elem_type ElemType;
```

Comment:
I'll make sure to write in the documentation that specifying `MatType` inside of `RangeSearchType` is the way to do it for DBSCAN... maybe it's a little awkward, but until we think of something better, it's better than having to specify `MatType` in two places. :+1: